lib/rules/space-infix-ops.js

```diff
@@ -39,97 +131,40 @@ module.exports = {
-         * @param {string} op - The operator of the main node
-         * @returns {Object} The violator token or null
-         * @private
-         */
-        function getFirstNonSpacedToken(left, right, op) {
-            const operator = sourceCode.getFirstTokenBetween(left, right, token => token.value === op);
-            const prev = sourceCode.getTokenBefore(operator);
-            const next = sourceCode.getTokenAfter(operator);
-
-            if (!sourceCode.isSpaceBetweenTokens(prev, operator) || !sourceCode.isSpaceBetweenTokens(operator, next)) {
-                return operator;
-            }
-
-            return null;
-        }
-
-        /**
-         * Reports an AST node as a rule violation
-         * @param {ASTNode} mainNode - The node to report
-         * @param {Object} culpritToken - The token which has a problem
-         * @returns {void}
-         * @private
-         */
-        function report(mainNode, culpritToken) {
-            context.report({
-                node: mainNode,
-                loc: culpritToken.loc.start,
-                message: "Operator '{{operator}}' must be spaced.",
-                data: {
-                    operator: culpritToken.value
-                },
-                fix(fixer) {
-                    const previousToken = sourceCode.getTokenBefore(culpritToken);
-                    const afterToken = sourceCode.getTokenAfter(culpritToken);
-                    let fixString = "";
-
-                    if (culpritToken.range[0] - previousToken.range[1] === 0) {
-                        fixString = " ";
-                    }
-
-                    fixString += culpritToken.value;
-
-                    if (afterToken.range[0] - culpritToken.range[1] === 0) {
-                        fixString += " ";
-                    }
-
-                    return fixer.replaceText(culpritToken, fixString);
-                }
-            });
-        }
-
-        /**
-         * Check if the node is binary then report
-         * @param {ASTNode} node node to evaluate
-         * @returns {void}
-         * @private
-         */
-        function checkBinary(node) {
-            const leftNode = (node.left.typeAnnotation) ? node.left.typeAnnotation : node.left;
-            const rightNode = node.right;
-
-            // search for = in AssignmentPattern nodes
-            const operator = node.operator || "=";
-
-            const nonSpacedNode = getFirstNonSpacedToken(leftNode, rightNode, operator);
-
-            if (nonSpacedNode) {
-                if (!(int32Hint && sourceCode.getText(node).endsWith("|0"))) {
-                    report(node, nonSpacedNode);
-                }
-            }
-        }
-
         /**
          * Check if the node is conditional
          * @param {ASTNode} node node to evaluate
          * @returns {void}
          * @private
          */
         function checkConditional(node) {
```

Comment:
If you like the gist of this PR, do you think these two functions need to be moved to outside as well?